kafka-ui-react-app/src/components/common/JSONEditor/JSONEditor.tsx

```diff
@@ -4,15 +4,21 @@ import 'ace-builds/src-noconflict/mode-json5';
 import 'ace-builds/src-noconflict/theme-textmate';
 import React from 'react';
 
-const JSONEditor: React.FC<IAceEditorProps> = (props) => (
-  <AceEditor
-    mode="json5"
-    theme="textmate"
-    tabSize={2}
-    width="100%"
-    wrapEnabled
-    {...props}
-  />
-);
+const JSONEditor: React.FC<IAceEditorProps> = (props) => {
+  const { readOnly, value } = props;
+  return (
+    <AceEditor
+      mode="json5"
+      theme="textmate"
+      tabSize={2}
+      width="100%"
+      height={
+        readOnly ? `${(value?.split('\n').length || 32) * 16}px` : '500px'
```

Comment:
Someone could use `readOnly` prop to disabled the editor while submitting the form. So please use another prop for determining the height. E.g. `mode: 'view' | 'edit'`.